src/ucp/wireup/wireup_ep.c

```diff
@@ -344,18 +347,32 @@ static ucs_status_t ucp_wireup_ep_check(uct_ep_h uct_ep, unsigned flags,
     if (wireup_ep->aux_ep != NULL) {
         ucs_assert(wireup_ep->aux_rsc_index != UCP_NULL_RESOURCE);
 
-        if (ucp_worker_iface(worker, wireup_ep->aux_rsc_index)->attr.cap.flags &
-            UCT_IFACE_FLAG_EP_CHECK) {
+        wiface = ucp_worker_iface(worker, wireup_ep->aux_rsc_index);
+        if (wiface->attr.cap.flags & UCT_IFACE_FLAG_EP_CHECK) {
             return uct_ep_check(wireup_ep->aux_ep, flags, comp);
         }
 
-        /* if EP_CHECK is not supported by auxiliary transport, it has to support
-         * a built-in keepalive mechanism to be able to detect peer failure during
-         * wireup
+        /* if EP_CHECK is not supported by auxiliary transport, it has to
+         * support a built-in keepalive mechanism to be able to detect peer
+         * failure during wireup
          */
-        ucs_assert(ucp_worker_iface(worker,
-                                    wireup_ep->aux_rsc_index)->attr.cap.flags &
-                   UCT_IFACE_FLAG_EP_KEEPALIVE);
+        ucs_assert(wiface->attr.cap.flags & UCT_IFACE_FLAG_EP_KEEPALIVE);
+        return UCS_OK;
+    }
+
+    if (tmp_ep != NULL && !(ucp_ep_config(tmp_ep)->key.ep_check_map != 0)) {
+        lane = ucs_ffs64_safe(wireup_ep->tmp_ep_check_map);
+        if (lane == 64) {
+            /* re-arm TMP EP check map */
+            wireup_ep->tmp_ep_check_map =
+                    ucp_ep_config(tmp_ep)->key.ep_check_map;
+
+            lane = ucs_ffs64_safe(wireup_ep->tmp_ep_check_map);
+            ucs_assert(lane != 64);
+        }
```

Comment:
can we unite code with regular keepalive?

Reply:
unfortunately, no
this regular KA is done for worker